### PR TITLE
(UI): 465 newsfeed

### PR DESCRIPTION
**Pull request by L03TJ3**
# Description

Adding newsfeed ui components

Todo:
- Awaiting desktop screen designs to finalize this pr

About # (link your issue here)
[#465](https://github.com/GoodDollar/GoodProtocolUI/issues/465)

# Screens
(mobile & desktop -- same size)
image

# How Has This Been Tested?
storybook
Please describe the tests that you ran to verify your changes.

# Checklist:
- [ ] PR title matches follow: (Feature|Bug|Chore) Task Name
- [ ] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: packages/good-design/src/apps/newsfeed/NewsFeed.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    pictureStyles,
+    titleStyles,
+    contentStyles,
+    footerStyles,
+    publishedStyles,
+    sponsoredStyles,
+    ...props
+  }) => {
+    const { picture, title, content, published, sponsored_logo } = item;
+    const formattedPublished = new Date(published).toLocaleDateString("en", {
```
Comment (johnsmith-gooddollar) [inline packages/good-design/src/apps/newsfeed/NewsFeed.tsx:33]: add use memo

File: packages/good-design/src/stories/apps/newsfeed/NewsFeed.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import React, { useContext } from "react";
+
+import { NewsFeed } from "../../../apps/newsfeed";
+
+import { NewsFeedContext, NewsFeedProvider } from "@gooddollar/web3sdk-v2";
+
+const devConfig = {
+  env: "qa",
+  ceramicConfig: {
```
Comment (johnsmith-gooddollar) [inline packages/good-design/src/stories/apps/newsfeed/NewsFeed.stories.tsx:9]: doesn't it splitted out yet ?

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

File: packages/good-design/src/apps/newsfeed/NewsFeed.tsx (hunk 1 above)
Comment (L03TJ3) [reply packages/good-design/src/apps/newsfeed/NewsFeed.tsx]: done

File: packages/good-design/src/stories/apps/newsfeed/NewsFeed.stories.tsx (hunk 2 above)
Comment (L03TJ3) [reply packages/good-design/src/stories/apps/newsfeed/NewsFeed.stories.tsx]: not relevant anymore

File: packages/good-design/src/theme/hoc/withTheme.tsx [hunk 3]
```diff
@@ -9,7 +9,8 @@ interface IWithThemeOpts {
 
 export const withTheme =
   (options?: IWithThemeOpts) =>
-  <T,>(Component: React.ComponentType<T>) => {
+  // @ts-ignore
+  <T,>(Component: React.ComponentType<T & { children?: React.ReactNode }> & { children?: React.ReactNode }) => {
```
Comment (johnsmith-gooddollar) [inline packages/good-design/src/theme/hoc/withTheme.tsx:13]: there should be generics like PropsWIthChildren etc. please look up how to describe it correctly using the types react provides

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

File: packages/good-design/src/apps/newsfeed/NewsFeed.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            <Text {...publishedStyles}>{formattedPublished}</Text>
+            {sponsored_logo && <SvgXml src={sponsored_logo} height="28" width="45" {...sponsoredStyles} />}
+          </HStack>
+        </CentreBox>
+      </CentreBox>
+    );
+  }
+);
+
+export const NewsFeed = ({ feed }: { feed: FeedPost[] }) => {
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/good-design/src/apps/newsfeed/NewsFeed.tsx]: ye this one noticed, just pushed it

File: packages/good-design/src/apps/newsfeed/NewsFeed.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    contentStyles,
+    footerStyles,
+    publishedStyles,
+    sponsoredStyles,
+    ...props
+  }) => {
+    const { picture, title, content, published, sponsored_logo } = item;
+
+    const formattedPublished = useMemo(() => {
+      return new Date(published).toLocaleDateString("en", {
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/good-design/src/apps/newsfeed/NewsFeed.tsx]: done

File: packages/good-design/src/core/images/SvgXml.native.tsx [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import React from "react";
+import { SvgXml as SVG } from "react-native-svg";
+
+const SvgXml = ({ src, width, height }: any) => <SVG xml={src} width={width} height={height} />;
+
+export default SvgXml;
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/good-design/src/core/images/SvgXml.native.tsx]: done

File: packages/good-design/src/core/images/SvgXml.tsx [hunk 7]
```diff
@@ -0,0 +1,6 @@
+import React from "react";
+import SVG from "react-inlinesvg";
+
+const SvgXml = ({ src, width, height }: any) => <SVG src={src} width={width} height={height} />;
+
+export default SvgXml;
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/good-design/src/core/images/SvgXml.tsx]: done

File: packages/good-design/src/theme/hoc/withTheme.tsx (hunk 3 above)
Comment (L03TJ3) [reply packages/good-design/src/theme/hoc/withTheme.tsx]: Ah that looks a lot cleaner yeah, updated now

**Review by johnsmith-gooddollar — APPROVED** [review]

**Pull request merged**